packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -117,14 +125,20 @@ async function ssrTransformScript(
     }
     const metadataStr = metadata ? `, ${JSON.stringify(metadata)}` : ''
 
-    // There will be an error if the module is called before it is imported,
-    // so the module import statement is hoisted to the top
-    s.appendLeft(
-      index,
+    s.update(
+      importNode.start,
+      importNode.end,
       `const ${importId} = await ${ssrImportKey}(${JSON.stringify(
         source,
       )}${metadataStr});\n`,
     )
+
+    // There will be an error if the module is called before it is imported,
+    // so the module import statement is hoisted to the top
+    if (importNode.start !== index) {
+      s.move(importNode.start, importNode.end, index)
+    }
```

Comment:
I'm testing something and would this work? :thinking: 

```ts
    if (hoistIndex === importNode.start) {
      // if first import is at the top, we can keep the position but move hoistIndex next to it.
      hoistIndex = importNode.end;
    } else {
      // push others at the same place
      s.move(importNode.start, importNode.end, hoistIndex)
    }
```

These are the test cases I tried:

<details>

```ts
test('multiple', async () => {
  expect(
    await ssrTransformSimpleCode(
      `\
import lib1 from "lib1"
import lib2 from "lib2"
import lib3 from "lib3"
`,
    ),
  ).toMatchInlineSnapshot(`
    "const __vite_ssr_import_0__ = await __vite_ssr_import__("lib1", {"importedNames":["default"]});
    const __vite_ssr_import_1__ = await __vite_ssr_import__("lib2", {"importedNames":["default"]});
    const __vite_ssr_import_2__ = await __vite_ssr_import__("lib3", {"importedNames":["default"]});



    "
  `)

  expect(
    await ssrTransformSimpleCode(
      `\
console.log("something else")
import lib1 from "lib1"
import lib2 from "lib2"
import lib3 from "lib3"
`,
    ),
  ).toMatchInlineSnapshot(`
    "const __vite_ssr_import_0__ = await __vite_ssr_import__("lib1", {"importedNames":["default"]});
    const __vite_ssr_import_1__ = await __vite_ssr_import__("lib2", {"importedNames":["default"]});
    const __vite_ssr_import_2__ = await __vite_ssr_import__("lib3", {"importedNames":["default"]});
    console.log("something else")



    "
  `)

  expect(
    await ssrTransformSimpleCode(
      `#!/usr/bin/env node
import lib1 from "lib1"
import lib2 from "lib2"
import lib3 from "lib3"
`,
    ),
  ).toMatchInlineSnapshot(`
    "#!/usr/bin/env node
    const __vite_ssr_import_0__ = await __vite_ssr_import__("lib1", {"importedNames":["default"]});
    const __vite_ssr_import_1__ = await __vite_ssr_import__("lib2", {"importedNames":["default"]});
    const __vite_ssr_import_2__ = await __vite_ssr_import__("lib3", {"importedNames":["default"]});



    "
  `)

  expect(
    await ssrTransformSimpleCode(
      `\
console.log(path)
import path from 'node:path'
console.log(fs)
import fs from 'node:fs'
`,
    ),
  ).toMatchInlineSnapshot(`
    "const __vite_ssr_import_0__ = await __vite_ssr_import__("node:path", {"importedNames":["default"]});
    const __vite_ssr_import_1__ = await __vite_ssr_import__("node:fs", {"importedNames":["default"]});
    console.log(__vite_ssr_import_0__.default)

    console.log(__vite_ssr_import_1__.default)

    "
  `)
})
```

Reply:
Oh, I was misunderstanding what was happening here. That should work.

Reply:
I thought we need to hoist re-export too, but I just noticed https://github.com/vitejs/vite/pull/14441/ stopped doing that to fix some cyclic import.